gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/MysqlMultiActiveLeaseArbiter.java

```diff
@@ -204,6 +212,14 @@ public MysqlMultiActiveLeaseArbiter(Config config) throws IOException {
     }
     initializeConstantsTable();
 
+    Thread retentionThread = new Thread(new Runnable() {
```

Comment:
rather than a sleeping/blocking thread, how about a scheduled thread pool executor taking this `Runnable` and an exec interval?

Reply:
changed